modules/common/route/types.go

```diff
@@ -37,4 +37,5 @@ type GenericRouteDetails struct {
 	ServiceName    string
 	TargetPortName string
 	FQDN           string
+	Annotation     map[string]string
```

Comment:
This should be Annotations (plural). The map could contain any number of annotation key / value pairs. That's also consistent with the `metav1.ObjectMeta{}`
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L230